simulation_ws/src/aws_robomaker_simulation_common/nodes/route_manager

```diff
@@ -65,38 +240,57 @@ class RouteManager():
             rospy.logerr("Route mode '%s' unknown, exiting route manager", self.route_mode)
             return
 
-        poses = rospy.get_param('~poses')
-        if not poses:
+        poses = rospy.get_param('~poses', [])
+        if not poses and self.route_mode!='dynamic':
             rospy.loginfo("Route manager initialized no goals, unable to route")
-
+        
         self.goals = RouteManager.route_modes[self.route_mode](poses)
-        rospy.loginfo("Route manager initialized with %s goals in %s mode", len(poses), self.route_mode)
+        rospy.loginfo("Route manager initialized in %s mode", self.route_mode)
+
+        self.bad_goal_counter = 0
+
 
     def to_move_goal(self, pose):
+        if pose is None:
+            raise ValueError("Goal position cannot be NULL")
+
         goal = MoveBaseGoal()
         goal.target_pose.header.stamp = rospy.Time.now()
         goal.target_pose.header.frame_id = "map"
         goal.target_pose.pose.position = Point(**pose['pose']['position'])
         goal.target_pose.pose.orientation = Quaternion(**pose['pose']['orientation'])
         return goal
 
+
     def route_forever(self):
         rate = rospy.Rate(1)
         while not rospy.is_shutdown():
             try:
+                if self.bad_goal_counter > 10:
+                    raise ValueError("Bad goal limit exceeded")
```

Comment:
Nitpick: I don't think this should be a `raise`, I think it should just be an `if-else`. You can move the `try` block below up to include `to_move_goal()` and add another `except` branch to it for `ValueError`.

Reply:
agreed, throwing error is not ideal behavior for this